packages/sources/view-starknet-latest-answer/src/transport/function.ts

```diff
@@ -0,0 +1,117 @@
+import { RpcProvider } from 'starknet'
+import { config } from '../config'
+
+const logger = makeLogger('View Starknet Latest Answer')
+
+export type StarknetLatestAnswerFunctionTransportTypes = BaseEndpointTypes
+
+type RequestParams = typeof inputParameters.validated
+
+export class StarknetLatestAnswerFunctionTransport extends SubscriptionTransport<StarknetLatestAnswerFunctionTransportTypes> {
+  provider: RpcProvider | undefined
+
+  async initialize(
+    dependencies: TransportDependencies<StarknetLatestAnswerFunctionTransportTypes>,
+    adapterSettings: StarknetLatestAnswerFunctionTransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+  }
+
+  async backgroundHandler(
+    context: EndpointContext<StarknetLatestAnswerFunctionTransportTypes>,
+    entries: RequestParams[],
+  ) {
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(param: RequestParams) {
+    let response: AdapterResponse<StarknetLatestAnswerFunctionTransportTypes['Response']>
+    try {
+      response = await this._handleRequestStarknet(param)
+    } catch (e: unknown) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: (e as AdapterInputError)?.statusCode || 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequestStarknet(
+    param: RequestParams,
+  ): Promise<AdapterResponse<StarknetLatestAnswerFunctionTransportTypes['Response']>> {
+    const { address } = param
+    const { STARKNET_RPC_URL } = config.settings
+
+    const rpcUrl = STARKNET_RPC_URL
+
+    if (!rpcUrl) {
+      throw new AdapterInputError({
+        statusCode: 400,
+        message: `Missing RPC URL: '${rpcUrl}' environment variables.`,
+      })
+    }
+
+    if (!this.provider) {
+      this.provider = new RpcProvider({ nodeUrl: rpcUrl })
+    }
+
+    const callData = {
+      contractAddress: address,
+      entrypoint: 'latest_round_data',
+      calldata: [],
+    }
+
+    const providerDataRequestedUnixMs = Date.now()
+
+    let res
+    try {
+      // res: [round_id, answer, block_num, started_at, updated_at]
+      res = await this.provider.callContract(callData)
+    } catch (e) {
+      throw new AdapterInputError({
+        statusCode: 502,
+        message: `RpcProvider.callContract Failed - ${e}`,
+      })
+    }
+    // extract field "answer"
+    const answer = res[1]
```

Comment:
I think we want to convert this into a decimal string. Could you use num.hexToDecimalString like here? https://github.com/smartcontractkit/external-adapters-js/pull/3311/files#diff-312ba1eba60bf3b6d7ac8004ffa0a0f76d13de5cae841c59bf23eeabee372ea3R63

Reply:
actually hold that thought -- we might actually need it in hex

Reply:
ok sorry for the confusion, we do need the hexadecimal value (in order to use ethabidecode later) so your current implementation is good!

Reply:
i'll update here when i find the answer

Reply:
Hex representations are very common in a lot of web3 libraries. I'd say it is an issue with the forward process if it can't handle the hex string personally. Just my 2 cents.

Reply:
ok so after my findings I learned we should indeed change it to decimals in order to simplify logic in the pipeline config. please use the num.hexToDecimalString method. thank you!

Reply:
Ok!